ch4/deepComparison.js

```diff
@@ -0,0 +1,42 @@
+const deepEqual = (value1, value2) => {
+  if (value1 === value2) return true;
+  if (value1 === null || value2 === null) 
```

Comment:
note that i forgot a return statement here and that makes a big difference because it means the checks of typeof value1 !== typeof value2 was getting skipped

Reply:
I added many tests and updated function